Add statistics function for #97

JBlond opened the pull request.

DigiLive commented on lib/jblond/Diff/Similarity.php:
```diff
@@ -223,4 +223,39 @@ private function ratioReduce(int $sum, array $triple): int
+        $return = [
+            'inserted'     => 0,
+            'deleted'      => 0,
+            'notModified'  => 0,
```
Array index is immediately overwritten before accessing

JBlond replied:
I don't understand what you mean.

DigiLive commented:
We have to think about what "Not Modified" means.
Currently it means all lines which aren't replaced or deleted. (White lines of base version)
Should a deleted/inserted line count as modified or not?

JBlond commented:
The white lines are not visible in an unfined diff. It might be safe to remove that from the stats.

DigiLive commented:
> 
> 
> The white lines are not visible in an unfined diff. It might be safe to remove that from the stats.

The stats should be independent from whats visible.
I was referring to them to clarify what I meant with "not modified".

We should define what "not modified" means.
Line...

1. is (partly) replaced. **n**!
2. is removed from base. y/**n**?
   (Is a completely removed line, a modified line?)
3. is added to new. y/**n**?
   (Doesn't exist in the base version, so can't be modified?)

In the above list, the bold selection is currently set.

JBlond commented:
I'm not satisfied with the counting.

The example/example.php output has 

```
Between the two versions, there's a 72% match.
Inserted lines: 1
Deleted lines: 1
Lines with replacement: 6
```

But it should be

Inserted lines: 1
Deleted lines: 2
Lines with replacement: 7

JBlond commented:
```
diff -u a.txt b.txt | diffstat                                                                                                     104ms [15:14]
 b.txt |   19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)
```

JBlond commented:
my bad, I just made a debug run, and that that it is "equal" in the switch rather than "notModified"

DigiLive commented on lib/jblond/Diff/Similarity.php:
```diff
@@ -223,4 +223,39 @@ private function ratioReduce(int $sum, array $triple): int
+        $return = [
+            'inserted' => 0,
+            'deleted'  => 0,
+            'equal'    => 0,
```
Array index is immediately overwritten at line 257 before accessing
